### PR TITLE
cothread

### PR DESCRIPTION
**Pull request by hhslepicka**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hhslepicka** [comment]
@conda-forge/help-python-c this one is ready to be merged.
Unfortunately the package does not provide a LICENSE file. 😞

File: recipes/cothread/0001-epicscorelibs.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+--- a/cothread/__init__.py
++++ b/cothread/__init__.py
+@@ -37,6 +37,10 @@ modules:
+     catools
+         Channel access support tools for client access to EPICS records.
+ '''
++try:
++    import epicscorelibs.path.cothread
++except ImportError:
++    pass
```
Comment (jakirkham) [inline recipes/cothread/0001-epicscorelibs.patch]: Why is this needed?

Comment (hhslepicka) [reply recipes/cothread/0001-epicscorelibs.patch]: @jakirkham this is needed to use the runtime dependency to libca provided by epicscorelibs.
Users can also have this library at some externally built location on their system so that is why it is optional and wrapped with the try except

Comment (jakirkham) [reply recipes/cothread/0001-epicscorelibs.patch]: Why not make a Conda package for that library as well?

Comment (hhslepicka) [reply recipes/cothread/0001-epicscorelibs.patch]: It is already on conda-forge... epicscorelibs.

Comment (xhochy) [reply recipes/cothread/0001-epicscorelibs.patch]: Then please use this as a run requirements. Conda packages should depend on all things that are required to run. Using libraries from somewhere else is not the default usage and should be avoided at all costs.

Comment (hhslepicka) [reply recipes/cothread/0001-epicscorelibs.patch]: @xhochy It is there already. https://github.com/conda-forge/staged-recipes/pull/9973/files#diff-0a6a7eb69a560925ee867b776505534aR30

Comment (xhochy) [reply recipes/cothread/0001-epicscorelibs.patch]: Why is this patch then still needed? (As you see if this is really needed, add a comment about this in the recipe so other people not ask again ;) )

Comment (hhslepicka) [reply recipes/cothread/0001-epicscorelibs.patch]: @xhochy this patch is needed because the python library looks at an environment variable for the location of the libca library. epicscorelibs provides this library and a wrapper for `cothread` to bypass this check. I will add a comment as suggested.
We use the same approach at other EPICS client libraries such as `pyepics` and `pyca`, all here at conda-forge.
